### ISSUE TITLE
Can't reproduce F1 scores

### ISSUE THREAD

**Issue by bobflagg** [lemonhu/NER-BERT-pytorch#2]
After training the model using the default parameters:
  python train.py --data_dir data/msra --bert_model_dir bert-base-chinese-pytorch --model_dir experiments/base_model
I ran evaluation with
  python evaluate.py --data_dir data/msra --bert_model_dir bert-base-chinese-pytorch --model_dir experiments/base_model 
and got the following results, which are very much lower than those described in the "detail results on the test set" from the repo README:
- Test metrics: loss: 00.05; f1: 58.29
             precision    recall  f1-score   support

        PER      35.36     90.09     50.79      1372
        ORG      35.24     85.70     49.94      1287
        LOC      53.90     92.11     68.01      2790

avg / total      44.83     90.09     59.41      5449
Any advice on how to reproduce the expected scores would be greatly appreciated.

**Comment by lemonhu** [comment]
Thank you for your feedback, sorry for replying to you for so long.

According to my experiment, the reason for your poor result should be the `pytorch-pretrained-bert ` version. When I try to use the latest version, it will have a very bad effect. When I switch to version 0.4.0, it will return to normal.

It is recommended to install a lower version of `pytorch-pretrained-bert` using `pip install pytorch-pretrained-bert==0.4.0`, and I will update the code later.

**Comment by lemonhu** [comment]
I re-clone the code again, the results of the first two epoch are as follows, [dependency version](https://github.com/lemonhu/NER-BERT-pytorch/blob/master/requirements.txt) and [default hyper-parameters](https://github.com/lemonhu/NER-BERT-pytorch/blob/master/experiments/base_model/params.json) has been updated.

100%|███████████████████████████| 1400/1400 [08:54<00:00, 2.72it/s, loss=0.023]

Train metrics: loss: 00.01; f1: 94.26
Val metrics: loss: 00.01; f1: 92.31
Found new best F1
Epoch 2/20
100%|███████████████████████████| 1400/1400 [08:33<00:00, 2.63it/s, loss=0.007]

Train metrics: loss: 00.00; f1: 96.90
Val metrics: loss: 00.01; f1: 94.16
Found new best F1

**Comment by bobflagg** [comment]
Thanks for the quick feedback. I've pulled the new version of the code and installed pytorch-pretrained-bert==0.4.0.  It's taking me longer than it should to verify the results because I keep running out of GPU memory (even after dropping the batch size) but will take another shot. Thanks again for your help!

**Comment by lemonhu** [comment]
Thanks for reporting, the problem is that the GPU memory is not enough. You need to reduce the size of the `batch_size` in the hyper-parameter configuration file [params.json](https://github.com/lemonhu/NER-BERT-pytorch/blob/master/experiments/base_model/params.json), such as, set `batch_size=8`.

**Comment by bobflagg** [comment]
Dropping to batch_size=8 did the trick.  The first three epochs are looking great:

Epoch 3/20
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████| 5250/5250 [37:59<00:00,  2.30it/s, loss=0.010]
- Train metrics: loss: 00.01; f1: 96.39
- Val metrics: loss: 00.02; f1: 91.76

Thanks for the help!

**Comment by bobflagg** [comment]
Here are my new evaluation results:

- Test metrics: loss: 00.02; f1: 90.97
             precision    recall  f1-score   support

        LOC      92.84     92.97     92.91      2790
        PER      92.42     93.29     92.85      1372
        ORG      82.54     87.41     84.91      1287

avg / total      90.30     91.74     91.00      5449

Much better!

**Comment by lemonhu** [comment]
Yeah, feel free to re-open if needed.
